pkg/kv/kvprober/kvprober.go

```diff
@@ -306,11 +308,42 @@ func (p *Prober) Metrics() Metrics {
 	return p.metrics
 }
 
+// LogLeaseholderInfo searches through the given tracing recording for log messages
+// indicating updated range information, extracts range IDs and leaseholder node IDs,
+// and logs this information only if the leaseholder for a range has been updated.
+func (p *Prober) LogLeaseholderInfo(ctx context.Context, recording tracingpb.Recording) {
```

Comment:
We need some tests of this new functionality. I suggest two tests:

1. Unit test of `LogLeaseholderInfo`. Prob the best way is to golden a trace in textual form, then parse the trace to get a `tracingpb.Recording`, then pass that into `LogLeaseholderInfo` and assert that expected leaseholder is parsed out of the trace. This test will run quickly, which is nice, but it could go out of date, if the trace format changes over time, which motivates the second test I have in mind...
2. Update one of the existing integration tests (https://github.com/cockroachdb/cockroach/blob/master/pkg/kv/kvprober/kvprober_integration_test.go) to assert that the leaseholder logging feature is working on an actual cluster. Here is a test that goldens out the logging done by `sqlprober`. Prob can follow same pattern here, tho details will be different, since different logging library: https://github.com/cockroachlabs/managed-service/pull/16806/files#diff-b7d1606dd13ab761d13bbd2d7785b9139bba43c958703e98d36dfb2bbaef9887R48

Happy to talk in person about the details here. Should be fun!

Reply:
Thank you! I will work on these once the actual code is working fine.